test/concatAll.js

```diff
@@ -0,0 +1,35 @@
+import { assert } from 'chai';
+import { NEL, Nil } from 'monet';
+
+import { concatAll } from '../src';
+import eq from './shared/eq';
+
+describe('concatAll', function() {
+  it('should concatenate arrays', function() {
+    eq(concatAll([[1], [2], [3]]), [1, 2, 3]);
+  });
+
+  it('should concatenate strings', function() {
+    eq(concatAll(['1', '2', '3']), '123');
+  });
+
+  it('should concatenate semigroups', function() {
+    eq(concatAll([NEL(1), NEL(2)]), NEL(1, NEL(2, Nil)));
+  });
+
+  it('should returns undefined if empty foldable was passed', function() {
+    eq(concatAll([]), undefined);
+  });
+
+  it('should throw if non-foldable is passed', function() {
+    assert.throws(() => concatAll(null), TypeError);
+  });
+
+  it('should throw if foldable contains non-semigroups', function() {
+    assert.throws(() => concatAll([1, 2, null, true]), TypeError);
+  });
+
+  it('should throw if foldable contains non-compatible semigroups', function() {
```

Comment:
This test is not passing on ramda 0.23.0

Reply:
Should I remove it then?

Reply:
We have to find out why it's not passing and do feature detection to enable/disable test depending on features R provides for us in particular version

Reply:
That version just doesn't throw for concating string and array, so I changed the test to try to concat array and string ;)

Reply:
Done.

Reply:
This fails in 0.21 tho!

I think I'll just remove the test.

Reply:
Ramda should be testing that in `R.concat` anyway.

Reply:
Just comment the test out, don't remove it. I'll play with that after merge.